build-logic-settings/environment/src/main/kotlin/gradlebuild/basics/BuildEnvironmentExtension.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gradlebuild.basics
+
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.provider.Property
+
+
+abstract class BuildEnvironmentExtension {
```

Comment:
💅 Why not an `interface`?

Reply:
I guess because I just copy/pasted the class. :) 
Changed it to `interface`.